examples/region/fairmq-start-ex-region-advanced-external.sh.in

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+export FAIRMQ_PATH=@FAIRMQ_BIN_DIR@
+
+transport="shmem"
+msgSize="1000000"
+
+if [[ $1 =~ ^[a-z]+$ ]]; then
+    transport=$1
+fi
+
+if [[ $2 =~ ^[0-9]+$ ]]; then
+    msgSize=$2
+fi
```

Comment:
if you want to keep positional arguments, just skip the checks and use [parameter substitution syntax](https://tldp.org/LDP/abs/html/parameter-substitution.html):

```suggestion
transport=${1:-shmem}
msgSize=${2:-1000000}
```

The checks sort of make the positional arguments optional, why not use real optionals then?

```suggestion
transport="shmem"
msgSize="1000000"

while getopts "t:s:" arg; do
  case $arg in
    t)
      transport=$OPTARG 
      ;;
    s)
      msgSize=$OPTARG
      ;;
  esac
done
```